app/models/health/ssm_saver.rb

```diff
@@ -25,16 +25,20 @@ def update
           # The CHA QA requires both the CHA and the SSM, so check both
           # also done in the ChaSaver so it can be done in either order
           @cha = @ssm.patient.recent_cha_form
-          if @cha.present? && @cha.completed? && @cha.reviewed?
-            qualifying_activity = setup_qualifying_activity
+          if @cha.present? && @cha.completed?
+            qualifying_activity = if @cha.reviewed?
+              setup_completed_qualifying_activity
+            else
+              setup_development_qualifying_activity
+            end
```

Comment:
So when you save an SSM, it checks to see if you have a CHA and then builds a QA based on that?  (I see the note now.)  You only get the QAs if you've completed both.  I feel like there's a race condition here, but I can't identify it.